src/ComputeSharp/Graphics/Extensions/Texture2DExtensions.cs

```diff
@@ -33,311 +60,467 @@ public static class Texture2DExtensions
     {
-        texture.CopyTo(destination, 0, 0, 0, 0, texture.Width, texture.Height);
+        source.CopyTo(destination, 0, 0, 0, 0, source.Width, source.Height);
     }
 
     /// <summary>
     /// Reads the contents of a <see cref="Texture2D{T}"/> instance and writes them into a target <see cref="ReadBackTexture2D{T}"/> instance.
     /// </summary>
     /// <typeparam name="T">The type of items stored on the texture.</typeparam>
-    /// <param name="texture">The input <see cref="Texture2D{T}"/> instance to read data from.</param>
+    /// <param name="source">The input <see cref="Texture2D{T}"/> instance to read data from.</param>
     /// <param name="destination">The target <see cref="ReadBackTexture2D{T}"/> instance to write data to.</param>
-    /// <param name="x">The horizontal offset in the source texture.</param>
-    /// <param name="y">The vertical offset in the source texture.</param>
+    /// <param name="sourceOffsetX">The horizontal offset in the source texture.</param>
+    /// <param name="sourceOffsetY">The vertical offset in the source texture.</param>
+    /// <param name="width">The width of the memory area to copy.</param>
+    /// <param name="height">The height of the memory area to copy.</param>
+    public static void CopyTo<T>(this Texture2D<T> source, ReadBackTexture2D<T> destination, int sourceOffsetX, int sourceOffsetY, int width, int height)
+        where T : unmanaged
+    {
+        source.CopyTo(destination, sourceOffsetX, sourceOffsetY, 0, 0, width, height);
+    }
+
+    /// <summary>
+    /// Reads the contents of a <see cref="Texture2D{T}"/> instance and writes them into a target <see cref="ReadBackTexture2D{T}"/> instance.
+    /// </summary>
+    /// <typeparam name="T">The type of items stored on the texture.</typeparam>
+    /// <param name="source">The input <see cref="Texture2D{T}"/> instance to read data from.</param>
+    /// <param name="destination">The target <see cref="ReadBackTexture2D{T}"/> instance to write data to.</param>
+    /// <param name="sourceOffsetX">The horizontal offset in the source texture.</param>
+    /// <param name="sourceOffsetY">The vertical offset in the source texture.</param>
+    /// <param name="destinationOffsetX">The horizontal offset in the destination texture.</param>
+    /// <param name="destinationOffsetY">The vertical offset in the destination texture.</param>
     /// <param name="width">The width of the memory area to copy.</param>
     /// <param name="height">The height of the memory area to copy.</param>
-    public static void CopyTo<T>(this Texture2D<T> texture, ReadBackTexture2D<T> destination, int x, int y, int width, int height)
+    public static void CopyTo<T>(this Texture2D<T> source, ReadBackTexture2D<T> destination, int sourceOffsetX, int sourceOffsetY, int destinationOffsetX, int destinationOffsetY, int width, int height)
         where T : unmanaged
     {
-        texture.CopyTo(destination, 0, 0, x, y, width, height);
```

Comment:
Was this a bug before or am I mis-mapping from old to new above?

Reply:
This is just the parameter being renamed, plus the swapped order.
We're now always doing, in order: source, destination, count.
Should make the APIs less error prone across overloads as they're more consistent 🙂